src/xtgeo/grid3d/grid_property.py

```diff
@@ -129,106 +129,140 @@ class GridProperty(_Grid3D):
 
        poronumpy = poro.values1d
 
-    Args:
-        *args: If a value exists, it should either be a file name, a Grid()
-            or a GridProperty() instance. See examples below
-        ncol (int): Number of columns.
-        nrow (int): Number of rows.
-        nlay (int): Number of layers.
-        values (numpy): A 3D masked numpy of shape (ncol, nrow, nlay).
-        name (str): Name of property.
-        discrete (bool): True if discrete property
-            (default is false).
-        fracture (bool): Indicates a fracture setup (for flow simulator)
-        codes (dict): A code to name dictionary (for discrete)
+    .. versionchanged:: 2.6 Possible to make GridProperty instance directly from Grid()
+    .. versionchanged:: 2.8 Possible to base it on existing GridProperty() instance
 
-    Alternatively, the same arguments as the from_file() method
-    can be used.
+    """
 
-    Returns:
-        A GridProperty object instance.
+    def __init__(
+        self,
+        pfile: Optional[Union[str, pathlib.Path, Any]] = None,
+        fformat: Optional[str] = "guess",
+        ncol: Optional[int] = 4,
+        nrow: Optional[int] = 3,
+        nlay: Optional[int] = 5,
+        name: Optional[str] = "unknown",
+        discrete: Optional[bool] = False,
+        date: Optional[str] = None,
+        grid: Optional[Any] = None,
+        linkgeometry: Optional[bool] = True,
+        fracture: Optional[bool] = False,
+        codes: Optional[dict] = None,
+        dualporo: Optional[bool] = False,
+        dualperm: Optional[bool] = False,
+        roxar_dtype: Optional[Any] = np.float32,
+        values: Optional[Union[np.ndarray, float, int]] = None,
+    ):
+        """Instantating.
```

Comment:
Elaborate?

Reply:
The elaboration is in the docstring for the Class.